src/binding/c/rma_api.txt

```diff
@@ -0,0 +1,466 @@
+    .seealso: MPI_Put
+    .impl: mpid
+    .earlyreturn: pt2pt_proc_null
+
+MPI_Win_allocate:
+    .desc: Create and allocate an MPI Window object for one-sided communication
+    .seealso: MPI_Win_allocate_shared MPI_Win_create MPI_Win_create_dynamic MPI_Win_free
+    .impl: mpid
+
+MPI_Win_allocate_shared:
+    .desc: Create an MPI Window object for one-sided communication and shared memory access, and allocate memory at each process
+    .seealso: MPI_Win_allocate MPI_Win_create MPI_Win_create_dynamic MPI_Win_free MPI_Win_shared_query
+    .impl: mpid
+
+MPI_Win_attach:
+    .desc: Attach memory to a dynamic window
+    .seealso: MPI_Win_create_dynamic MPI_Win_detach
+    .impl: mpid
+{ -- early_return --
+    if (size == 0)
+        goto fn_exit;
+}
+
+MPI_Win_complete:
+    .desc: Completes an RMA operations begun after an MPI_Win_start
+    .impl: mpid
+
+MPI_Win_create:
+    .desc: Create an MPI Window object for one-sided communication
+    .seealso: MPI_Win_allocate MPI_Win_allocate_shared MPI_Win_create_dynamic MPI_Win_free
+    .impl: mpid
+/*
+    Notes:
+    
+    The displacement unit argument is provided to facilitate address arithmetic in
+    RMA operations: the target displacement argument of an RMA operation is scaled
+    by the factor disp_unit specified by the target process, at window creation.
+    
+    The info argument provides optimization hints to the runtime about the expected
+    usage pattern of the window. The following info keys are predefined.
+    
+    . no_locks - If set to true, then the implementation may assume that passive
+        target synchronization (i.e., 'MPI_Win_lock', 'MPI_Win_lock_all') will not be used on
+        the given window. This implies that this window is not used for 3-party
+        communication, and RMA can be implemented with no (less) asynchronous agent
+        activity at this process.
+    
+    . accumulate_ordering - Controls the ordering of accumulate operations at the
+        target.  The argument string should contain a comma-separated list of the
+        following read/write ordering rules, where e.g. "raw" means read-after-write:
+        "rar,raw,war,waw".
+    
+    . accumulate_ops - If set to same_op, the implementation will assume that all
+        concurrent accumulate calls to the same target address will use the same
+        operation. If set to same_op_no_op, then the implementation will assume that
+        all concurrent accumulate calls to the same target address will use the same
+        operation or 'MPI_NO_OP'. This can eliminate the need to protect access for
+        certain operation types where the hardware can guarantee atomicity. The default
+        is same_op_no_op.
+*/
+
+MPI_Win_create_dynamic:
+    .desc: Create an MPI Window object for one-sided communication
+    .seealso: MPI_Win_attach MPI_Win_detach MPI_Win_allocate MPI_Win_allocate_shared MPI_Win_create MPI_Win_free
+    .impl: mpid
+/*
```

Comment:
In the autogen file of `win_create_dynamic.c`, why the input parameter is `- comm - intra-communicator (handle)`? We don't specify intra or inter in any other RMA files.

Reply:
I think `MPIR_Err_return_win` should be `MPIR_Err_return_comm`  at `fn_fail` in `win_create_dynamic.c`, because the win_ptr is not created yet.
Same issue in `win_create.c`.